### PR TITLE
Changed all language class names and added READMEs where they where missing

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
This pull request solves part of #983

In particular :
- Language modules should provide readme files that explain the design rationales of the token extraction
- The language facades should not be named the same (API-breaking change!!!)

I tried to replace all textual references to the Language classes, although I might have missed some.

### CODE REVIEW

**Review by Kr0nox — APPROVED** [review]

File: cli/src/test/java/de/jplag/cli/MetamodelDynamicMetamodelLanguageTest.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 import de.jplag.Language;
 
-class LanguageTest extends CommandLineInterfaceTest {
+class MetamodelDynamicMetamodelLanguageTest extends CommandLineInterfaceTest {
```
Comment (tsaglam) [inline cli/src/test/java/de/jplag/cli/MetamodelDynamicMetamodelLanguageTest.java:13]: Probably renaming gone wrong?

File: README.md [hunk 2]
```diff
@@ -109,7 +109,7 @@ The new API makes it easy to integrate JPlag's plagiarism detection into externa
 <!-- To assure that the code example is always correct, it must be kept in sync
 with [`ReadmeCodeExampleTest#testReadmeCodeExample`](core/src/test/java/de/jplag/special/ReadmeCodeExampleTest.java). -->
 ```java
-Language language = new de.jplag.java.Language();
+Language language = new de.jplag.java.JavaLanguage();
```
Comment (tsaglam) [inline README.md:112]: Now we do not need the FQN anymore!

File: docs/1.-How-to-Use-JPlag.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ The new API makes it easy to integrate JPlag's plagiarism detection into externa
 **Example:**
 
 ```java
-Language language = new de.jplag.java.Language();
+Language language = new de.jplag.java.JavaLanguage();
```
Comment (tsaglam) [inline docs/1.-How-to-Use-JPlag.md:56]: Now we do not need the FQN anymore!

File: core/src/test/java/de/jplag/special/ReadmeCodeExampleTest.java [hunk 4]
```diff
@@ -25,7 +26,7 @@ class ReadmeCodeExampleTest {
      */
     @Test
     void testReadmeCodeExample() {
-        Language language = new de.jplag.java.Language();
+        Language language = new JavaLanguage();
```
Comment (tsaglam) [inline core/src/test/java/de/jplag/special/ReadmeCodeExampleTest.java:29]: Every change here should always be mirrored in the readme.

File: languages/emf-metamodel-dynamic/README.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ This module is based on the EMF dependencies available on maven central. These m
 For the token extraction, we visit the containment tree of the metamodel and extract tokens for all metamodel elements based on their concrete metaclass. In this module, we thus extract tokens based on a dynamic token set.
 
 ### Usage
-To use this module, add the `-l emf-metamodel-dynamic` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.emf.dynamic.Language()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
+To use this module, add the `-l emf-metamodel-dynamic` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.emf.dynamic.DynamicMetamodelLanguage()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
```
Comment (tsaglam) [inline languages/emf-metamodel-dynamic/README.md:12]: Use DynamicEmfLanguage for the dynamic module, and EmfLanguage for the other one.

File: languages/emf-metamodel-dynamic/src/main/java/de/jplag/emf/dynamic/DynamicMetamodelLanguage.java [hunk 6]
```diff
@@ -8,7 +9,7 @@
  * created token set instead of a hand-picked one.
  * @author Timur Saglam
  */
-public class Language extends de.jplag.emf.Language { // currently not included in the CLI
+public class DynamicMetamodelLanguage extends MetamodelLanguage { // currently not included in the CLI
```
Comment (tsaglam) [inline languages/emf-metamodel-dynamic/src/main/java/de/jplag/emf/dynamic/DynamicMetamodelLanguage.java:12]: Use DynamicEmfLanguage for the dynamic module, and EmfLanguage for the other one.

File: languages/emf-metamodel/README.md [hunk 7]
```diff
@@ -9,7 +9,7 @@ This module is based on the EMF dependencies available on maven central. These m
...
 
 ### Usage
-To use this module, add the `-l emf-metamodel` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.emf.Language()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
+To use this module, add the `-l emf-metamodel` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.emf.MetamodelLanguage()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
```
Comment (tsaglam) [inline languages/emf-metamodel/README.md:12]: EmfLanguage

File: languages/emf-metamodel/src/main/java/de/jplag/emf/MetamodelLanguage.java [hunk 8]
```diff
@@ -16,7 +16,7 @@
  * @author Timur Saglam
  */
 @MetaInfServices(de.jplag.Language.class)
-public class Language implements de.jplag.Language {
+public class MetamodelLanguage implements de.jplag.Language {
```
Comment (tsaglam) [inline languages/emf-metamodel/src/main/java/de/jplag/emf/MetamodelLanguage.java:19]: EmfLanguage

File: languages/emf-model/src/main/java/de/jplag/emf/model/ModelLanguage.java [hunk 9]
```diff
@@ -14,13 +15,13 @@
  * @author Timur Saglam
  */
 @MetaInfServices(de.jplag.Language.class)
-public class Language extends de.jplag.emf.dynamic.Language {
+public class ModelLanguage extends DynamicMetamodelLanguage {
```
Comment (tsaglam) [inline languages/emf-model/src/main/java/de/jplag/emf/model/ModelLanguage.java:18]: Use EmfModelLanguage here.

File: languages/rlang/src/main/java/de/jplag/rlang/RlangLanguage.java [hunk 10]
```diff
@@ -13,15 +13,15 @@
  * This represents the R language as a language supported by JPlag.
  */
 @MetaInfServices(de.jplag.Language.class)
-public class Language implements de.jplag.Language {
+public class RlangLanguage implements de.jplag.Language {
```
Comment (tsaglam) [inline languages/rlang/src/main/java/de/jplag/rlang/RlangLanguage.java:16]: Just RLanguage here.

File: languages/rlang/README.md [hunk 11]
```diff
@@ -13,4 +13,4 @@ The choice of tokens is based directly on the CodeGra-de version, whereas the ex
 Like in other modules, e.g. for Java and C#, the tokens account for the beginning and the end of control flow structures, for control flow keywords, and some kinds of expressions. As R is very different from other programming languages in JPlag, it remains to be seen whether the R module can hold up to the others.
 
 ### Usage
-To use the R module, add the `-l R` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.rlang.Language()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
\ No newline at end of file
+To use the R module, add the `-l R` flag in the CLI, or use a `JPlagOption` object with `new de.jplag.rlang.RlangLanguage()` as `language` in the Java API as described in the usage information in the [readme of the main project](https://github.com/jplag/JPlag#usage) and [in the wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).
```
Comment (tsaglam) [inline languages/rlang/README.md:16]: Just RLanguage here.

File: languages/text/src/main/java/de/jplag/text/PlainTextLanguage.java [hunk 12]
```diff
@@ -15,12 +15,12 @@
  * but there are better approaches for text plagiarism out there (based on NLP techniques).
  */
 @MetaInfServices(de.jplag.Language.class)
-public class Language implements de.jplag.Language {
+public class PlainTextLanguage implements de.jplag.Language {
```
Comment (tsaglam) [inline languages/text/src/main/java/de/jplag/text/PlainTextLanguage.java:18]: Use NaturalLanguage here.

**Review by tsaglam — COMMENTED** [review]
Just a few minor comments regarding my preferred naming.

**Comment by tsaglam** [comment]
@TwoOfTwelve as discussed, only the conflicts are left, then I can merge.

**Comment by tsaglam** [comment]
@TwoOfTwelve just as a reminder.

**Pull request merged**